dev/io.openliberty.microprofile.reactive.messaging.internal_fat.common/src/com/ibm/ws/microprofile/reactive/messaging/fat/AppValidator.java

```diff
@@ -0,0 +1,202 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.fat;
+
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+
+import com.ibm.websphere.simplicity.PropertiesAsset;
+import com.ibm.websphere.simplicity.RemoteFile;
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.matchers.Matchers;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Helper class for building and testing apps that should fail to start in a particular way
+ * <p>
+ * Example:
+ *
+ * <pre>
+ * {@code
+ * AppValidator.validateAppOn(SHARED_SERVER)
+ *                 .withClass(MyInvalidClass.class)
+ *                 .failsWith("CWMFT5001E")
+ *                 .run();
+ * }
+ * </pre>
+ *
+ * <p>
+ * This creates a web application containing just {@code MyInvalidClass}, deploys it, checks that it fails to start and that a message containing {@code CWMFT5001E} is found in the
```

Comment:
given this can do both successful and failure scenarios, I would correct the comment about it only being for failures